### ISSUE TITLE
VIP Rewards

### ISSUE THREAD

**Issue by DanielCaspers** [DanielCaspers/CA-Portal#18]
**Motivation**
Customers should be able to view their rewards and what they can redeem with their rewards.

**Acceptance criteria**
- [x] 1. A user shall be able to view
- [x] 1.1 Current rewards point balance
- [x] 1.2 ~~Points remaining until the next milestone~~ De-scoped as all of the data is hard-coded in the UI
- [x] 1.3 What can be redeemed with those points

- [x] 2. The information shall be exposed from one or more MurphyAuto APIs.

**Descoped requirements**
The following requirements were redacted by the customer.

- [ ] 1.4. Point credits/debits list
- [ ] 1.4.1 The list shall contain
- [ ] 1.4.1.1 Transaction Date
- [ ] 1.4.1.2 Points credited/debited
- [ ] 1.4.2 The table shall be permanently sorted in reverse chronological order.

**Comment by DanielCaspers** [comment]
@scaspers remind me how to log in again, and let me know if you guys have other ideas for criteria. I felt like point 3 was a feature add to further game-ify their points earning.

**Comment by YoMurphy** [comment]
Customers often ask or wonder if they received their rewards. We currently indicate past redemption's in our emails but should there be some level of tracking or a points earned table? Maybe just Service Date and Points earned? Could be shown in detail similar to a bank account with Debits and Credits.

**Comment by DanielCaspers** [comment]
@scaspers @YoMurphy Do you think it is important to have history back as far as possible, or would seeing say the latest X credits/redemptions be enough?

I would suggest considering showing 5 to 10 of the latest credit redemptions or credits in a table within a widget on the homepage. It could provide client side pagination if you find it necessary.

**Comment by DanielCaspers** [comment]
@YoMurphy @scaspers I've updated the issue description. Please respond with "approve" if you think this is all we need. 

As currently specified between #18 and #22, points reward balance and points to next milestone will be duplicated in both the homepage widget and in this separate view.

**Comment by DanielCaspers** [comment]
@scaspers @YoMurphy I'm blocked on proceeding further with this feature until

1) There is additional API support
2) We come to a decision on how we'd like to represent their points credits/debits as well as what can be redeemed with what they have. 

I wanted to remind you of the proof of concept UI I did for this awhile back to show you where we are. The points powering the speedometer graph are from the API, but the rest is still faked awaiting points table description via an API if you're interested. 

![MyCar - Murphy Automotive (1)](https://user-images.githubusercontent.com/6056563/69907563-aad2f780-139c-11ea-9293-fdb53ea72272.gif)

**Comment by DanielCaspers** [comment]
Updated from offline decisions with @scaspers 

1) @YoMurphy  will look at what exists and determine if want to keep the points table as it is?
2) Steve and Rick will confirm whether or not we cut the speedo graph.

**Comment by YoMurphy** [comment]
The VIP page seems confusing to me for a few reasons. 
1. The point gauge will constantly change and doesn't represent its value very well. It would be my vote to remove it.
2. The check marks next to rewards would make me think I've earned each of them, so I'd also think i would get each of them. I'd suggest keeping our schedule simple and not dynamic. Remove colors and icons.
3. The oil change schedule may not even get noticed in a second tab. Maybe just include it in a list on the same page in a second group. Would create more scrolling, but prevent it from being missed.

As you can tell I'm leaning towards clarity and less glitz for this page. Keep it simple and clear.

**Comment by DanielCaspers** [comment]
@YoMurphy @scaspers 
, how do you feel about interleaving the two lists and display lowest to highest number of points and let the user do the math? 

I only suggest it since scrolling can be missed too if a user stops scrolling if they think they don't have enough points to get what they think they can use, and side by side like in the existing public infographic is not a strong option at this size. 

I'm fine with it either way; lmk when you've reached consensus and I'll wrap it up!

**Comment by YoMurphy** [comment]
I feel like merging the oil change with the other rewards will clutter up our list and create even more confusion. Maybe display the service and wash rewards first, then oil change second. What do you think @scaspers ?

**Comment by scaspers** [comment]
I agree that they should be separated but in one "page".  Maybe the current tabs could be links to anchor tags for those that want to quickly scroll to that section?

**Comment by DanielCaspers** [comment]
@scaspers @YoMurphy , the view has been simplified as requested.

1) No more speedometer graph
2) Point total nice and big
3) No more check marks, x's, or colors to indicate whether or not anything has been earned.
4) Separate sections for oil change and service + car wash rewards that can be scrolled to. 

The points debit/credit list is not included. AFAIK there exists no such API for that yet. **Please advise whether or not we need the points debit/credit list for completing acceptance criteria.** 

![MyCar - Murphy Automotive (3)](https://user-images.githubusercontent.com/6056563/70872175-b3662900-1f6b-11ea-9ed1-f5deeec520f7.gif)

**Comment by scaspers** [comment]
We will leave transaction data off for now but thought the display could be tabular something like:
![image](https://user-images.githubusercontent.com/12421599/73866274-2e301f80-480a-11ea-8e9e-b4827bf1812c.png)

**Comment by DanielCaspers** [comment]
New look for company number 002. 
![image](https://user-images.githubusercontent.com/6056563/73905843-4df53100-4866-11ea-92b9-f2c76f0b5058.png)
